lzy/execution-env/src/main/java/ai/lzy/env/base/DockerEnvironment.java

```diff
@@ -291,16 +299,66 @@ private void prepareImage(String image, LogStream out) throws Exception {
         var msg = "Pulling image %s ...".formatted(image);
         LOG.info(msg);
         out.log(msg);
+        Set<String> allowedPlatforms = config.allowedPlatforms();
         AtomicInteger pullingAttempt = new AtomicInteger(0);
-        retry.executeCallable(() -> {
+        try (var pullResponseItem = retry.executeCallable(() -> {
             LOG.info("Pulling image {}, attempt {}", image, pullingAttempt.incrementAndGet());
-            final var pullingImage = client
-                .pullImageCmd(config.image())
-                .exec(new PullImageResultCallback());
-            return pullingImage.awaitCompletion();
-        });
+            if (allowedPlatforms.isEmpty()) {
+                return pullWithPlatform(image, null);
+            } else {
+                for (String platform : config.allowedPlatforms()) {
+                    try {
+                        return pullWithPlatform(image, platform);
+                    } catch (DockerClientException e) {
+                        String exceptionMessage = e.getMessage();
+                        if (exceptionMessage.contains(NO_MATCHING_MANIFEST_ERROR) ||
+                                exceptionMessage.contains(NOT_MATCH_PLATFORM_ERROR)) {
+                            LOG.info("Cannot find image = {} for platform = {}: message = {}",
+                                    image, platform, exceptionMessage);
+                        } else {
+                            throw e;
+                        }
+                    }
+                }
+            }
+            return null;
+        })
+        ) {
```

Comment:
`)` should be on line 325

Reply:
Only ) without { ? I mean 
}))
{
...
right?
or
})) {

Reply:
first case